src/aws-cpp-sdk-core/include/aws/core/client/AWSClientEventStreamingAsyncTask.h

```diff
@@ -125,6 +132,92 @@ class AWS_CORE_LOCAL BidirectionalEventStreamingTask final {
   std::shared_ptr<Aws::Utils::Threading::Semaphore> m_sem;
 };
 
+/**
+ * Smithy-compatible bi-directional streaming task for modern AWS clients
+ */
+template <typename OutcomeT, typename ClientT, typename RequestT, typename HandlerT>
+class AWS_CORE_LOCAL SmithyBidirectionalStreamingTask final {
```

Comment:
maybe we want to put this in the smithy namespace and not in core as this is the sra client specific